File: MMORPG_2/Assets/Script/Data/LocalData/Create/GameLevelMonsterEntity.cs

//===================================================
//作    者：
//创建时间：2019-05-18 11:12:46
//备    注：此代码为工具生成 请勿手工修改
//===================================================
using System.Collections;

/// <summary>
/// GameLevelMonster实体
/// </summary>
public partial class GameLevelMonsterEntity : AbstractEntity
{
    /// <summary>
    /// 所属章节
    /// </summary>
    public int Chapter { get; set; }

    /// <summary>
    /// 所属章节关卡Id
    /// </summary>
    public int GameLevel { get; set; }

    /// <summary>
    /// 难度等级
    /// </summary>
    public int Grade { get; set; }

    /// <summary>
    /// 区域Id
    /// </summary>
    public int RegionId { get; set; }

    /// <summary>
    /// 怪物Id
    /// </summary>
    public int MonsterId { get; set; }

    /// <summary>
    /// 怪物数量
    /// </summary>
    public int MonsterCount { get; set; }

    /// <summary>
    /// 掉落经验
    /// </summary>
    public int DropExp { get; set; }

    /// <summary>
    /// 掉落金币
    /// </summary>
    public int DropCoin { get; set; }

    /// <summary>
    /// 掉落装备
    /// </summary>
    public string[][] DropEquip { get; set; }

    /// <summary>
    /// 掉落材料
    /// </summary>
    public string[][] DropMaterial { get; set; }

}

File: GameServer/GameServerApp/GameServerApp/Data/LocalData/CreateLua/GameLevelMonsterEntity.lua
GameLevelMonsterEntity = { Id = 0, Chapter = 0, GameLevel = 0, Grade = 0, RegionId = 0, MonsterId = 0, MonsterCount = 0, DropExp = 0, DropGold = 0, DropEquip = 0, DropMaterial = 0 }

--这句是重定义元表的索引，就是说有了这句，这个才是一个类
GameLevelMonsterEntity.__index = GameLevelMonsterEntity;

function GameLevelMonsterEntity.New(Id, Chapter, GameLevel, Grade, RegionId, MonsterId, MonsterCount, DropExp, DropGold, DropEquip, DropMaterial)
    local self = { }; --初始化self
    setmetatable(self, GameLevelMonsterEntity); --将self的元表设定为Class

    self.Id = Id;
    self.Chapter = Chapter;
    self.GameLevel = GameLevel;
    self.Grade = Grade;
    self.RegionId = RegionId;
    self.MonsterId = MonsterId;
    self.MonsterCount = MonsterCount;
    self.DropExp = DropExp;
    self.DropGold = DropGold;
    self.DropEquip = DropEquip;
    self.DropMaterial = DropMaterial;

    return self;
end